src/Sarif/Visitors/ExtractAllArtifactLocationsVisitor.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+
+    public class ExtractAllArtifactLocationsVisitor : SarifRewritingVisitor
+    {
+        private Run _currentRun;
+        public HashSet<ArtifactLocation> allArtifactLocations { get; private set; }
+
+        public ExtractAllArtifactLocationsVisitor()
+        {
+            allArtifactLocations = new HashSet<ArtifactLocation>(ArtifactLocation.ValueComparer);
```

Comment:
>allArtifactLocations [](start = 12, length = 20)

It's not clear to me whether you want to reset this at the start of each run (at the start of `VisitRun`) or accumulate across runs, as the code stands. You'll need to consider the expected use case, and whatever the answer, you'll need to document it in the class summary comment.

Reply:
Currently, I expect to get all locations in a given set of runs.  I'll add the documentation confirming this.  I'd be curious if you see the reset version being the more common use case, outside of the bug filer.

---
In reply to: [386741455](https://github.com/microsoft/sarif-sdk/pull/1794#discussion_r386741455) [](ancestors = 386741455)